src/@next/components/atoms/LanguageProvider/LanguageProvider.tsx

```diff
@@ -0,0 +1,16 @@
+import { I18nProvider } from "@lingui/react";
+import React from "react";
+import { I18nContext } from "../I18nLoader";
+import { IProps } from "./types";
+
+export const LanguageProvider: React.FC<IProps> = ({ children }: IProps) => {
+  const { language, catalogs } = React.useContext(I18nContext);
```

Comment:
These 2 providers (I18nLoader and LanguageProvider) seem to be always used together and adhere to the following order I18nLoader (parent) => LanguageProvider (child). What if we unified these two and made it as one component, the functionality would stay the same, less code (we wouldn't have to use `React.useContext` in LanguageProvider but instead get these straight off the component state) and also makes it more reusable as one doesn't need to rely on another already existing in the tree. What do you think?

Reply:
Sounds like a good idea, but I will stay with `React.useContext`, which allows us to change the language from any place in the application.

Reply:
We could still do that, since we keep the context we can use React.useContext (or even export a util hook called `useLanguage` that wraps `React.useContext(I18nContext)`) - we won't need to use context inside LanguageProvider since the values we are fetching for already exist in the state (if merged into one HoC)